### PR TITLE
Add docs for Auth, Auth.Cookies, Auth.Certificate

### PR DESCRIPTION
**Pull request by pranavkm**
Contributes to https://github.com/dotnet/aspnetcore/issues/26397

### CODE REVIEW

File: src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs [hunk 1]
```diff
@@ -108,52 +106,51 @@ private static bool IsAjaxRequest(HttpRequest request)
...
-        /// Implements the interface method by invoking the related delegate method.
+        /// Invoked during sign in.
         /// </summary>
-        /// <param name="context"></param>
+        /// <param name="context">The <see cref="CookieSigningInContext"/>.</param>
         public virtual Task SigningIn(CookieSigningInContext context) => OnSigningIn(context);
 
         /// <summary>
-        /// Implements the interface method by invoking the related delegate method.
+        /// Invoked after sign in is completed.
```
Comment (HaoK) [inline src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs:121]:
```suggestion
        /// Invoked after sign in has completed.
```

File: src/Security/Authentication/Core/src/AuthenticationHandler.cs [hunk 2]
```diff
@@ -11,34 +11,73 @@
...
+        /// <summary>
+        /// Gets the <see cref="HttpResponse" /> associated with the current request.
+        /// </summary>
         protected HttpResponse Response
         {
             get => Context.Response;
         }
 
+        /// <summary>
+        /// Gets the path that triggered authentication.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/AuthenticationHandler.cs:54]:
```suggestion
        /// Gets the path as seen by the authentication middleware.
```

File: src/Security/Authentication/Core/src/AuthenticationHandler.cs [hunk 3]
```diff
@@ -11,34 +11,73 @@
...
+        /// </summary>
         protected PathString OriginalPath => Context.Features.Get<IAuthenticationFeature>()?.OriginalPath ?? Request.Path;
 
+        /// <summary>
+        /// Gets the path base that triggered authentication.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/AuthenticationHandler.cs:59]:
```suggestion
        /// Gets the path base as seen by the authentication middleware.
```

File: src/Security/Authentication/Core/src/Events/RemoteAuthenticationContext.cs [hunk 4]
```diff
@@ -42,8 +42,16 @@ public abstract class RemoteAuthenticationContext<TOptions> : HandleRequestConte
         /// </summary>
         public void Success() => Result = HandleRequestResult.Success(new AuthenticationTicket(Principal!, Properties, Scheme.Name));
 
+        /// <summary>
+        /// Calls <see cref="HandleRequestResult.Fail(Exception)"/> indicating that authentication failed.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/Events/RemoteAuthenticationContext.cs:46]: That's an implementation detail that's not relevant to the caller.
```suggestion
        /// Indicates that authentication failed.
```

File: src/Security/Authentication/Core/src/Events/RemoteAuthenticationContext.cs [hunk 5]
```diff
@@ -42,8 +42,16 @@ public abstract class RemoteAuthenticationContext<TOptions> : HandleRequestConte
...
+        /// </summary>
+        /// <param name="failure">The exception associated with the failure.</param>
         public void Fail(Exception failure) => Result = HandleRequestResult.Fail(failure);
 
+        /// <summary>
+        /// Calls <see cref="HandleRequestResult.Fail(string)"/> indicating that authentication failed.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/Events/RemoteAuthenticationContext.cs:52]:
```suggestion
        /// Indicates that authentication failed.
```

File: src/Security/Authentication/Core/src/ISecureDataFormat.cs [hunk 6]
```diff
@@ -5,15 +5,41 @@
 
 namespace Microsoft.AspNetCore.Authentication
 {
+    /// <summary>
+    /// A ccontract for securing data.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/ISecureDataFormat.cs:9]:
```suggestion
    /// A contract for securing data.
```

File: src/Security/Authentication/Core/src/PolicySchemeHandler.cs [hunk 7]
```diff
@@ -15,21 +15,33 @@ namespace Microsoft.AspNetCore.Authentication
     /// </summary>
     public class PolicySchemeHandler : SignInAuthenticationHandler<PolicySchemeOptions>
     {
+        /// <summary>
+        /// Initializes a new isntance of <see cref="PolicySchemeHandler"/>.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/PolicySchemeHandler.cs:19]:
```suggestion
        /// Initializes a new instance of <see cref="PolicySchemeHandler"/>.
```

File: src/Security/Authentication/Core/src/SignInAuthenticationHandler.cs [hunk 8]
```diff
@@ -16,9 +15,17 @@ namespace Microsoft.AspNetCore.Authentication
     public abstract class SignInAuthenticationHandler<TOptions> : SignOutAuthenticationHandler<TOptions>, IAuthenticationSignInHandler
         where TOptions : AuthenticationSchemeOptions, new()
     {
+        /// <summary>
+        /// Initializes a new isntance of <see cref="SignInAuthenticationHandler{TOptions}"/>.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/SignInAuthenticationHandler.cs:19]:
```suggestion
        /// Initializes a new instance of <see cref="SignInAuthenticationHandler{TOptions}"/>.
```

File: src/Security/Authentication/Core/src/SignOutAuthenticationHandler.cs [hunk 9]
```diff
@@ -15,9 +14,17 @@ namespace Microsoft.AspNetCore.Authentication
     public abstract class SignOutAuthenticationHandler<TOptions> : AuthenticationHandler<TOptions>, IAuthenticationSignOutHandler
         where TOptions : AuthenticationSchemeOptions, new()
     {
+        /// <summary>
+        /// Initializes a new isntance of <see cref="SignOutAuthenticationHandler{TOptions}"/>.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/SignOutAuthenticationHandler.cs:18]:
```suggestion
        /// Initializes a new instance of <see cref="SignOutAuthenticationHandler{TOptions}"/>.
```

File: src/Security/Authentication/Core/src/TextEncoder.cs [hunk 10]
```diff
@@ -3,6 +3,9 @@
 
 namespace Microsoft.AspNetCore.Authentication
 {
+    /// <summary>
+    /// Allows encoding and decoding base-64 text.
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/TextEncoder.cs:7]:
```suggestion
    /// Allows encoding and decoding base-64 url encoded text.
```

File: src/Security/Authentication/Core/src/TicketSerializer.cs [hunk 11]
```diff
@@ -66,6 +79,11 @@ public virtual void Write(BinaryWriter writer, AuthenticationTicket ticket)
             PropertiesSerializer.Default.Write(writer, ticket.Properties);
         }
 
+        /// <summary>
+        /// Wrutes
```
Comment (Tratcher) [inline src/Security/Authentication/Core/src/TicketSerializer.cs:83]: ??

File: src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs [hunk 12]
```diff
@@ -9,17 +9,60 @@
...
 namespace Microsoft.Extensions.DependencyInjection
 {
+    /// <summary>
+    /// Extension methods to configure JWT bearer authentication.
+    /// </summary>
     public static class JwtBearerExtensions
     {
+        /// <summary>
+        /// Configures the <see cref="AuthenticationBuilder"/> to perform JWT-bearer authentication using the default scheme.
+        /// The default scheme is provided by <see cref="JwtBearerDefaults.AuthenticationScheme"/>.
```
Comment (Tratcher) [inline src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs:19]:
```suggestion
        /// Enables JWT-bearer authentication using the default scheme <see cref="JwtBearerDefaults.AuthenticationScheme"/>.
```

File: src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs [hunk 13]
```diff
@@ -9,17 +9,60 @@
...
+        /// </para>
+        /// </summary>
+        /// <param name="builder">The <see cref="AuthenticationBuilder"/>.</param>
+        /// <returns>A reference to <paramref name="builder"/> after the operation has completed.</returns>
         public static AuthenticationBuilder AddJwtBearer(this AuthenticationBuilder builder)
             => builder.AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, _ => { });
 
+        /// <summary>
+        /// Configures the <see cref="AuthenticationBuilder"/> to perform JWT-bearer authentication using the default scheme.
+        /// The default scheme is provided by <see cref="JwtBearerDefaults.AuthenticationScheme"/>.
```
Comment (Tratcher) [inline src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs:31]:
```suggestion
        /// Enables JWT-bearer authentication using the default scheme <see cref="JwtBearerDefaults.AuthenticationScheme"/>.
```

File: src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs [hunk 14]
```diff
@@ -9,17 +9,60 @@
...
+        /// </para>
+        /// </summary>
+        /// <param name="builder">The <see cref="AuthenticationBuilder"/>.</param>
+        /// <param name="configureOptions">A delegate that allows configuring <see cref="JwtBearerOptions"/>.</param>
+        /// <returns>A reference to <paramref name="builder"/> after the operation has completed.</returns>
         public static AuthenticationBuilder AddJwtBearer(this AuthenticationBuilder builder, Action<JwtBearerOptions> configureOptions)
             => builder.AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, configureOptions);
 
+        /// <summary>
+        /// Configures the <see cref="AuthenticationBuilder"/> to perform JWT-bearer authentication using the specified scheme.
```
Comment (Tratcher) [inline src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs:43]:
```suggestion
        /// Enables JWT-bearer authentication using the specified scheme.
```

File: src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs [hunk 15]
```diff
@@ -9,17 +9,60 @@
...
+        /// </summary>
+        /// <param name="builder">The <see cref="AuthenticationBuilder"/>.</param>
+        /// <param name="authenticationScheme">The authenticatio nscheme.</param>
+        /// <param name="configureOptions">A delegate that allows configuring <see cref="JwtBearerOptions"/>.</param>
+        /// <returns>A reference to <paramref name="builder"/> after the operation has completed.</returns>
         public static AuthenticationBuilder AddJwtBearer(this AuthenticationBuilder builder, string authenticationScheme, Action<JwtBearerOptions> configureOptions)
             => builder.AddJwtBearer(authenticationScheme, displayName: null, configureOptions: configureOptions);
 
+        /// <summary>
+        /// Configures the <see cref="AuthenticationBuilder"/> to perform JWT-bearer authentication using the specified scheme.
```
Comment (Tratcher) [inline src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs:56]:
```suggestion
        /// Enables JWT-bearer authentication using the specified scheme.
```

File: src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs [hunk 16]
```diff
@@ -9,17 +9,60 @@
...
             => builder.AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, configureOptions);
 
+        /// <summary>
+        /// Configures the <see cref="AuthenticationBuilder"/> to perform JWT-bearer authentication using the specified scheme.
+        /// <para>
+        /// JWT bearer authentication performs authentication by extracting and validating a JWT token from the <c>Authorization</c> request header.
+        /// </para>
+        /// </summary>
+        /// <param name="builder">The <see cref="AuthenticationBuilder"/>.</param>
+        /// <param name="authenticationScheme">The authenticatio nscheme.</param>
```
Comment (Tratcher) [inline src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs:49]:
```suggestion
        /// <param name="authenticationScheme">The authentication scheme.</param>
```

File: src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs [hunk 17]
```diff
@@ -9,17 +9,60 @@
...
             => builder.AddJwtBearer(authenticationScheme, displayName: null, configureOptions: configureOptions);
 
+        /// <summary>
+        /// Configures the <see cref="AuthenticationBuilder"/> to perform JWT-bearer authentication using the specified scheme.
+        /// <para>
+        /// JWT bearer authentication performs authentication by extracting and validating a JWT token from the <c>Authorization</c> request header.
+        /// </para>
+        /// </summary>
+        /// <param name="builder">The <see cref="AuthenticationBuilder"/>.</param>
+        /// <param name="authenticationScheme">The authenticatio nscheme.</param>
```
Comment (Tratcher) [inline src/Security/Authentication/JwtBearer/src/JwtBearerExtensions.cs:62]:
```suggestion
        /// <param name="authenticationScheme">The authentication scheme.</param>
```

**Review by Tratcher — APPROVED** [review]

File: src/Security/Authentication/Core/src/TicketSerializer.cs (hunk 11 above)
Comment (pranavkm) [reply src/Security/Authentication/Core/src/TicketSerializer.cs:83]:
```suggestion
        /// Wrutes 
```
```suggestion
        /// Writes the specified <paramref name="ClaimsIdentity" />.
```

File: src/Security/Authentication/Core/src/TicketSerializer.cs [hunk 18]
```diff
@@ -66,6 +79,11 @@ public virtual void Write(BinaryWriter writer, AuthenticationTicket ticket)
             PropertiesSerializer.Default.Write(writer, ticket.Properties);
         }
 
+        /// <summary>
+        /// Wrutes 
+        /// </summary>
+        /// <param name="writer"></param>
```
Comment (pranavkm) [inline src/Security/Authentication/Core/src/TicketSerializer.cs:85]:
```suggestion
        /// <param name="writer">The <see cref="BinaryWriter" />.</param>
```

File: src/Security/Authentication/Core/src/TicketSerializer.cs [hunk 19]
```diff
@@ -66,6 +79,11 @@ public virtual void Write(BinaryWriter writer, AuthenticationTicket ticket)
...
+        /// <param name="identity"></param>
```
Comment (pranavkm) [inline src/Security/Authentication/Core/src/TicketSerializer.cs:86]:
```suggestion
        /// <param name="identity">The <see cref="ClaimsIdentity" />.</param>
```

File: src/Security/Authentication/Core/src/TicketSerializer.cs [hunk 20]
```diff
@@ -66,6 +79,11 @@ public virtual void Write(BinaryWriter writer, AuthenticationTicket ticket)
             PropertiesSerializer.Default.Write(writer, ticket.Properties);
         }
 
+        /// <summary>
+        /// Writes the specified <paramref name="ClaimsIdentity" />.
```
Comment (pranavkm) [inline src/Security/Authentication/Core/src/TicketSerializer.cs:83]:
```suggestion
        /// Writes the specified <paramref name="identity" />.
```

**Comment by pranavkm** [comment]
@dotnet/aspnet-build merge please

**Pull request merged**